classes/Cart.php

```diff
@@ -1914,7 +1914,7 @@ public function getOrderTotal(
             return 0;
         }
         if ($virtual && $type == Cart::BOTH) {
-            $type = Cart::BOTH_WITHOUT_SHIPPING;
+            $type = Cart::BOTH;
```

Comment:
The whole `if` statement is actually unnecessary.

Reply:
The fix works but if the discount is 100% you can not place the order, the button "Order with an obligation to pay" still appear and disabled 
https://image.prntscr.com/image/viS6zjKXQJCoX8h1uAV85g.png

Reply:
Hi @pandisoft thanks for the feedback.

I'm going to add a few unit tests following @Quetzacoalt91 recommandations in order to cover all of these usecases so we know we're in safe waters 😄 I'll add this usecase then